go/vt/vtgate/evalengine/integration/string_fun_test.go

```diff
@@ -204,3 +204,35 @@ func TestBuiltinRepeat(t *testing.T) {
 
 	}
 }
+
+func TestBuiltinConv(t *testing.T) {
+	var conn = mysqlconn(t)
+	defer conn.Close()
+	cases := []string{
+		"10",
+		"10 + '10' + 10",
+		"-10",
+		"'10'",
+	}
+	bases := []string{
+		"-1",
+		"1",
+		"2",
+		"4",
+		"8",
+		"10",
+		"16",
+		"32",
+	}
```

Comment:
This is a very limited set of test cases. You need to widen this. Particularly important: hex literals.